src/mixins/canvas_events.mixin.js

```diff
@@ -141,6 +144,19 @@
 
     /**
      * @private
+     * @param {Event} e Event object fired on mouseenter
+     */
+    _onMouseEnter: function(e) {
+      if (!this.findTarget(e)) {
+        var target = this._hoveredTarget || null;
+        this.fire('mouse:over', { target: target, e: e });
+        this._hoveredTarget = null;
+        target && target.fire('mouseenter', { e: e });
```

Comment:
i think ```mouseenter``` is our ```mouseover```.

But beside that, if you do not find any target, ```this.findTarget(e)``` why do you want to fire the event for the last hoverdTarget ?